dpnp/dpnp_iface_indexing.py

```diff
@@ -493,28 +493,32 @@ def indices(
     return res
 
 
-def nonzero(x, /):
+def nonzero(a):
     """
     Return the indices of the elements that are non-zero.
 
+    Returns a tuple of arrays, one for each dimension of `a`,
+    containing the indices of the non-zero elements in that
+    dimension. The values in `a` are always tested and returned in
+    row-major, C-style order.
+
     For full documentation refer to :obj:`numpy.nonzero`.
 
+    Parameters
+    ----------
+    a : {dpnp.ndarray, usm_ndarray}
+        Input array.
+
     Returns
     -------
     out : tuple[dpnp.ndarray]
         Indices of elements that are non-zero.
 
-    Limitations
-    -----------
-    Parameters `x` is supported as either :class:`dpnp.ndarray`
-    or :class:`dpctl.tensor.usm_ndarray`.
-    Otherwise the function will be executed sequentially on CPU.
-    Input array data types are limited by supported DPNP :ref:`Data types`.
-
     See Also
     --------
     :obj:`dpnp.flatnonzero` : Return indices that are non-zero in
                               the flattened version of the input array.
+    :obj:`dpnp.ndarray.nonzero` : Equivalent ndarray method.
     :obj:`dpnp.count_nonzero` : Counts the number of non-zero elements
                                 in the input array.
 
```

Comment:
In the Notes section, it is better to use `a` for the variable for consistency. (feel free to fix it in another PR since this one has passed CI)

``a[nonzero(a)]``, it is recommended to use ``a[a.astype(bool)]`` or ``a[a != 0]``

Reply:
Good catch, thank you!